File: app/src/main/kotlin/com/foodshare/features/activity/domain/repository/ActivityRepository.kt
package com.foodshare.features.activity.domain.repository

import com.foodshare.features.activity.domain.model.ActivityItem
import com.foodshare.features.activity.domain.model.PostActivityItem
import com.foodshare.features.activity.domain.model.PostActivityStats
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for activity data operations.
 *
 * SYNC: Mirrors Swift ActivityRepository
 */
interface ActivityRepository {

    /**
     * Fetch activities with pagination.
     */
    suspend fun getActivities(
        offset: Int = 0,
        limit: Int = 20
    ): Result<List<ActivityItem>>

    /**
     * Observe activities as a real-time flow.
     */
    fun observeActivities(): Flow<ActivityItem>

    /**
     * Get activities for a specific post.
     */
    suspend fun getPostActivities(
        postId: Int,
        limit: Int = 50
    ): Result<List<PostActivityItem>>

    /**
     * Get recent activities for the current user's posts.
     */
    suspend fun getRecentActivitiesForMyPosts(): Result<List<PostActivityItem>>

    /**
     * Get activity stats for a specific post.
     */
    suspend fun getPostActivityStats(postId: Int): Result<PostActivityStats>
}

File: Sources/FoodShare/Features/Activity/Domain/Repositories/ActivityRepository.swift
//
//  ActivityRepository.swift
//  Foodshare
//
//  Repository protocol for activity feed operations
//

import Foundation

/// Repository for managing activity feed data
protocol ActivityRepository: Sendable {
    /// Fetch recent activities with pagination
    /// - Parameters:
    ///   - offset: Starting position for pagination
    ///   - limit: Number of items to fetch
    /// - Returns: Array of activity items
    func fetchActivities(offset: Int, limit: Int) async throws -> [ActivityItem]

    /// Fetch cached activities if available
    /// - Returns: Cached activities or empty array
    func fetchCachedActivities() async -> [ActivityItem]

    /// Cache activities for offline access
    /// - Parameter activities: Activities to cache
    func cacheActivities(_ activities: [ActivityItem]) async throws
}
